src/pybind/mgr/volumes/fs/volume.py

```diff
@@ -424,6 +424,27 @@ def subvolume_getpath(self, fs_handle, **kwargs):
             ret = self.volume_exception_to_retval(ve)
         return ret
 
+    @connection_pool_wrap
+    def list_subvolumes(self, fs_handle, **kwargs):
+        ret        = 0, "", ""
+        groupname  = kwargs['group_name']
+
+        try:
+            with SubVolume(self.mgr, fs_handle) as sv:
+                spec = SubvolumeSpec(None, groupname)
+                if not self.group_exists(sv, spec):
+                    raise VolumeException(
+                        -errno.ENOENT, "Subvolume group '{0}' not found".format(groupname))
+                path = sv.get_group_path(spec)
+                # When default subvolume group is not yet created we just return an empty list.
+                if path is None:
+                    return 0, '[]', ""
```

Comment:
minor nit: assign this to `ret` (+ introduce `else` section) so as to have a single point of return.